### ISSUE TITLE
Increase GPU usage

### ISSUE THREAD

**Issue by kTeo-0** [nipkownix/re4_tweaks#250]
### Describe the feature you'd like to see added

If i'm not mistaken, this game is almost entirely restricted to use CPU for rendering everything and that may be the biggest problem with slowmotion. Not sure but IF i'm right about that, Is there a way to increase GPU usage?

**Comment by emoose** [comment]
It does make use of DX9 for rendering (which uses GPU), but it doesn't really make use of it "natively", the game code is still calling into GX APIs which were used on Gamecube/Wii, except the X360 port team added a layer which translates the GX calls over to DirectX9 - probably adding a bit of CPU overhead in the process.

Maybe possible to optimize those calls a little, but doubt it'd help that much really... I did notice they used a flag when initing DX9 which slowed down FPS a lot though, but removing that flag makes things pretty unstable (the flag is meant for increased multi-thread stability): https://github.com/nipkownix/re4_tweaks/issues/44

Haven't been able to improve stability with it much yet, but maybe worth looking into some more...:

**Comment by FutonGama** [comment]
Its not possible change to Vulkan? AMD users have reported to me have problems with this game. I think Vulkan can be great for both platforms.

**Comment by emoose** [comment]
You could maybe try dxvk to make it run on Vulkan instead, IIRC nipkow did mention it improving performance a little, maybe could help increase GPU usage/framerate for others too, but doubt it'd help with CPU usage much (game is already going from GX->DX9->GPU, dxvk would add yet another layer to it...)

https://www.reddit.com/r/pcgaming/comments/mlfcsc/a_guide_to_dxvk_on_windows/ seems like a decent guide for how to set it up, haven't tried myself yet though (middle of reinstalling Windows atm)

There's not much chance of us being able to change the native gfx API the game uses though, besides not having that much experience with 3D APIs, there's also thousands of different gfx calls used in the game, without source code patching/hooking all that would be a huge task...

(though one thing I'd like to look into is updating it from DX9 to DX9Ex, might be able to help with the `Direct3DDevice9::Present` freezes with #44 with that since PresentEx adds some extra flags... tried doing some tests but didn't have much luck with it yet though)

**Comment by kTeo-0** [comment]
>tried #255 and my game didn't open :\

**Comment by emoose** [comment]
Huh darn, seems to work fine for me with 1.0.6, does setting `MultithreadFix` to false allow game to start for you?

Maybe worth trying to start the EXE with `MultithreadFix = true` a few times, might be a random launch issue, remember having those a few times with this stuff before...

**Comment by kTeo-0** [comment]
> 

nope, still can't get anything running. No crashlog either. Weird...

**Comment by emoose** [comment]
Added a new framelimiter which seems to reduce CPU usage quite a bit, you can find it here, would appreciate any testers for it!
https://github.com/nipkownix/re4_tweaks/issues/44#issuecomment-1179453070

>nope, still can't get anything running. No crashlog either. Weird...

Even with MultithreadFix = false it fails to start too? If so that is strange, maybe something else that we changed recently is breaking it, could you try the build from https://github.com/nipkownix/re4_tweaks/actions/runs/2629839609 too?

**Comment by kTeo-0** [comment]
> Added a new framelimiter which seems to reduce CPU usage quite a bit, you can find it here, would appreciate any testers for it! #257 
#257 is giving me the same results: game doesn't even start, and there's no crashlog.

> Even with MultithreadFix = false it fails to start too? If so that is strange, maybe something else that we changed recently is breaking it, could you try the build from https://github.com/nipkownix/re4_tweaks/actions/runs/2629839609 too?
alright i have no idea what is going on here, but none of the builds i'm trying is giving me any different results.

by the way, i've found a dll that makes the game run on Vulkan, and so far all the lag is entirely gone. Areas that previously were lagging all the way down to 30fps are now on a smooth 60fps!

This dll has been working flawlessly for everyone that i asked to test it EXCEPT @FutonGama, but i think it could have something to do with him having previously been using reshade (it's the same d3d9.dll, maybe it writes something else in the exe or something).

**Comment by emoose** [comment]
Hmm, did you try it without dxvk installed? Could be something incompatible with it, besides that maybe worth checking if the last published 1.7.6 release works? https://github.com/nipkownix/re4_tweaks/releases/tag/1.7.6

>and so far all the lag is entirely gone. Areas that previously were lagging all the way down to 30fps are now on a smooth 60fps!
Wow nice, didn't expect dxvk to work that well, interesting... will have to try testing to make sure all our re4_tweaks stuff works fine with it.

**Comment by lAlexl7l** [comment]
So i try this https://github.com/nipkownix/re4_tweaks/pull/257 but the game doesnt even start it crashes at the begining does this one requires something special to amke it run?

**Comment by emoose** [comment]
@lAlexl7l nah it should be running without needing anything else, could you try seeing if changing the `ReplaceFramelimiter` setting helps at all?

If that doesn't help, maybe try the https://github.com/nipkownix/re4_tweaks/releases/tag/1.7.6 release and see if that works with your setup.

**Comment by lAlexl7l** [comment]
1.7.6 version works but it lags on me and crashes on some areas this one it just crashes on start game window doenst even show up i did put ReplaceFramelimiter to true and still happends

**Comment by nipkownix** [comment]
Just updated the test build in #257.
If anyone can test a bit more, that'd be nice. Thanks!

**Comment by nipkownix** [comment]
Also, btw:

>tried https://github.com/nipkownix/re4_tweaks/pull/255 and my game didn't open :\

I'm pretty sure that was happening because you were trying to use a debug version, not a release one :p
Not sure why, though.

I've been linking the release version now to avoid confusion.

**Comment by nipkownix** [comment]
@FutonGama @kTeo-0 This builds adds Vulkan support + ``MultithreadFix``: 
https://github.com/nipkownix/re4_tweaks/suites/9403429268/artifacts/443786662

Could you guys check if your performance usage improves with it?
